res/app/device-list/details/device-list-details-directive.js

```diff
@@ -277,6 +279,20 @@ module.exports = function DeviceListDetailsDirective(
         }
       })()
 
+      // check if new column needs a new scope or not
+      // and build accordingly
+      function buildColumn(col, device) {
```

Comment:
Yes, definitely we want to avoid having paging as long as possible.
One of the goals is to make STF to feel more like a realtime application rather than a web page.

So if possible:
- Try to optimise as much with using plain DOM and just updating the changes. Theoretically this should work.
- Next worst case: use virtual-dom
- Next next-worst case: use virtual scrolling
- Next-next-next worst case: use paging